GetEmissionHistory.cs

```diff
@@ -0,0 +1,173 @@
+using System.Net;
+using FirebaseAdmin.Auth;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using MySqlConnector;
+using NoCO2.Util;
+using Company.Function;
+
+namespace NoCO2.Function
+{
+  public class GetEmissionHistory
+  {
+    static GetEmissionHistory()
+    {
+      FirebaseInitializer.Initialize();
+    }
+
+    private const double EMISSION_GOAL = 7.36;
+
+    [Function("GetEmissionHistory")]
+    public async Task<HttpResponseData> GetEmissionHistoryWithUserKey(
+        [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "get-emission-history")] HttpRequestData req)
+    {
+      var responseBodyObject = new {
+        reply = "InternalError"
+      };
+      try
+      {
+        req.Body.TryParseJson<GeneralUserKeyBody>(out var requestBody);
+
+        // Get "UserKey" parameter from HTTP request as either parameter or post value
+        string userKey = requestBody?.UserKey;
+        UserRecord userRecord = await FirebaseAuth.DefaultInstance.GetUserAsync(userKey);
+        int matchedUserID = GetUserIdIfUserKeyExistsInDB(userKey);
+
+        if (matchedUserID == -1)
+        {
+          // There is no user that has a matching hashed userkey from input userkey
+          responseBodyObject = new {
+            reply = "UserNotFound"
+          };
+          return await HttpResponseDataFactory.GetHttpResponseData(req, HttpStatusCode.BadRequest, responseBodyObject);
+        }
+
+        // Format the list of emissions into an object for HttpResponseData
+        List<DailyEmission> emissionHistory = await GetDailyEmissionsForUserWithinOneYear(matchedUserID);
+        var successResponseBodyObject = new {
+            reply = "Success",
+            History = emissionHistory
+        };
+        return await HttpResponseDataFactory.GetHttpResponseData(req, HttpStatusCode.OK, successResponseBodyObject);
+      } catch (ArgumentException) {
+        responseBodyObject = new {
+          reply = "InvalidArgument"
+        };
+        return await HttpResponseDataFactory.GetHttpResponseData(req, HttpStatusCode.BadRequest, responseBodyObject);
+      } catch (FirebaseAuthException) {
+        responseBodyObject = new {
+          reply = "UserKeyNotAuth"
+        };
+        return await HttpResponseDataFactory.GetHttpResponseData(req, HttpStatusCode.BadRequest, responseBodyObject);
+      } catch (Exception) {
+        return await HttpResponseDataFactory.GetHttpResponseData(req, HttpStatusCode.InternalServerError, responseBodyObject);
+      }
+      throw new NotImplementedException();
+    }
+
+    private int GetUserIdIfUserKeyExistsInDB(string userKey) {
```

Comment:
We should separate this code and make it another class, this is repeated code within this class and SubmitUserActivites, and all that need to verify userKey which are most, if not all functions. I can do this after this code is pushed. Also will simplify testing and reduce redundancy.